types/src/crypto/asymmetric_key.rs

```diff
@@ -209,8 +184,12 @@ impl Clone for SecretKey {
     fn clone(&self) -> Self {
         match self {
             SecretKey::System => SecretKey::System,
-            SecretKey::Ed25519(sk) => Self::ed25519(*sk.as_bytes()),
-            SecretKey::Secp256k1(sk) => Self::secp256k1(*sk),
+            SecretKey::Ed25519(sk) => Self::ed25519_from_bytes(*sk.as_bytes()).unwrap(),
+            SecretKey::Secp256k1(sk) => {
+                let raw_bytes: [u8; SECP256K1_SECRET_KEY_LENGTH] =
+                    sk.to_bytes().as_slice().try_into().unwrap();
+                Self::secp256k1_from_bytes(&raw_bytes).unwrap()
```

Comment:
`k256::ecdsa::SigningKey` doesn't implement `Clone` according to [the docs][1].

We should consider following suite. 

[1]: https://docs.rs/k256/0.7.2/k256/ecdsa/struct.SigningKey.html

Reply:
(in a separate ticket)